isaaclab_arena/tests/test_sequential_open_door.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) 2025, The Isaac Lab Arena Project Developers (https://github.com/isaac-sim/IsaacLab-Arena/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import gymnasium as gym
+import torch
+
+from isaaclab_arena.tests.utils.subprocess import run_simulation_app_function
+
+NUM_STEPS = 1
+HEADLESS = True
+
+
+def get_test_environment(remove_reset_door_state_event: bool, num_envs: int):
+    """Returns a scene which we use for these tests."""
+
+    from isaaclab_arena.assets.asset_registry import AssetRegistry
+    from isaaclab_arena.cli.isaaclab_arena_cli import get_isaaclab_arena_cli_parser
+    from isaaclab_arena.embodiments.franka.franka import FrankaEmbodiment
+    from isaaclab_arena.environments.arena_env_builder import ArenaEnvBuilder
+    from isaaclab_arena.environments.isaaclab_arena_environment import IsaacLabArenaEnvironment
+    from isaaclab_arena.scene.scene import Scene
+    from isaaclab_arena.tasks.open_door_task import OpenDoorTask
+    from isaaclab_arena.tasks.sequential_task_base import SequentialTaskBase
+    from isaaclab_arena.utils.pose import Pose
+
+    class SequentialOpenDoorTask(SequentialTaskBase):
+        def __init__(
+            self,
+            openable_object,
+            subtasks,
+            episode_length_s=None,
+        ):
+            super().__init__(subtasks=subtasks, episode_length_s=episode_length_s)
+            self.openable_object = openable_object
+
+        def get_metrics(self):
+            return []
+
+        def get_mimic_env_cfg(self, embodiment_name: str):
+            return None
+
+    args_parser = get_isaaclab_arena_cli_parser()
+    args_cli = args_parser.parse_args(["--num_envs", str(num_envs)])
+
+    asset_registry = AssetRegistry()
+    background = asset_registry.get_asset_by_name("packing_table")()
+    microwave = asset_registry.get_asset_by_name("microwave")()
+
+    # Put the microwave on the packing table.
+    microwave.set_initial_pose(
+        Pose(
+            position_xyz=(0.6, -0.00586, 0.22773),
+            rotation_wxyz=(0.7071068, 0, 0, -0.7071068),
+        )
+    )
+
+    subtask_1 = OpenDoorTask(microwave)
+    subtask_2 = OpenDoorTask(microwave)
+
+    scene = Scene(assets=[background, microwave])
+
+    isaaclab_arena_environment = IsaacLabArenaEnvironment(
+        name="sequential_open_door",
+        embodiment=FrankaEmbodiment(),
+        scene=scene,
+        task=SequentialOpenDoorTask(microwave, [subtask_1, subtask_2]),
+    )
+
+    env_builder = ArenaEnvBuilder(isaaclab_arena_environment, args_cli)
+    name, cfg = env_builder.build_registered()
+    if remove_reset_door_state_event:
+        cfg.events.reset_door_state_subtask_0 = None
+        cfg.events.reset_door_state_subtask_1 = None
```

Comment:
Is this because we don't want the environment to reset before the test can check the condition? Suggestion to add a comment indicating why we zero these out.